From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 30 Oct 2024 13:44:56 -0500
Subject: [PATCH] Fix wayland window associations by renaming .desktop file

For Wayland window managers to correctly associate open windows with the corresponding .desktop file the name of the `.desktop` file sans `.desktop` suffix needs to match the window appId.

By default Qt sets the appId to the name of the binary, but `QCoreApplication::setOrganizationDomain("seafile.com");` in `src/main.cpp` causes it to use the domain to create a fully qualified appId (`com.seafile.seafile-applet`). This does not match the desktop file `seafile.desktop`, which causes open windows to not be correctly associated with the launcher. While it's possible to use `QGuiApplication::setDesktopFileName()` to override the appId it's more consistent with FreeDesktop specs to use the fully qualified appId instead.

To see the difference this makes it is sufficient to start seafile-applet in a Plasma Wayland session (which has easily accessible window property introspection tools) with and without `com.seafile.seafile-applet.desktop` being the name of the `.desktop` file. Without the window will show the generic Wayland icon in the overview, with it will show the seafile icon shipped in the package.

Resolves haiwen/seafile-client#1568 (probably)
---
 CMakeLists.txt                                               | 2 +-
 data/{seafile.desktop => com.seafile.seafile-applet.desktop} | 0
 debian/seafile-gui.install                                   | 2 +-
 3 files changed, 2 insertions(+), 2 deletions(-)
 rename data/{seafile.desktop => com.seafile.seafile-applet.desktop} (100%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5474977..7768c26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -745,7 +745,7 @@ install(DIRECTORY
 )
 
 install(FILES
-  ${CMAKE_SOURCE_DIR}/data/seafile.desktop
+  ${CMAKE_SOURCE_DIR}/data/com.seafile.seafile-applet.desktop
   DESTINATION share/applications
 )
 
diff --git a/data/seafile.desktop b/data/com.seafile.seafile-applet.desktop
similarity index 100%
rename from data/seafile.desktop
rename to data/com.seafile.seafile-applet.desktop
diff --git a/debian/seafile-gui.install b/debian/seafile-gui.install
index 29143cc..8419f90 100644
--- a/debian/seafile-gui.install
+++ b/debian/seafile-gui.install
@@ -1,4 +1,4 @@
 usr/bin/seafile-applet
 usr/share/pixmaps/seafile.png
-usr/share/applications/seafile.desktop
+usr/share/applications/com.seafile.seafile-applet.desktop
 usr/share/icons/hicolor/*/apps/seafile.png
